helix-core/src/test/java/org/apache/helix/integration/TestCleanupExternalView.java

```diff
@@ -109,18 +109,14 @@ public void test() throws Exception {
     // System.out.println("re-enabling controller");
     admin.enableCluster(clusterName, true);
 
-    ExternalView externalView = null;
-    for (int i = 0; i < 10; i++) {
-      Thread.sleep(100);
-      externalView = accessor.getProperty(keyBuilder.externalView("TestDB0"));
-      // System.out.println("externalView: " + externalView);
-      if (externalView == null) {
-        break;
-      }
-    }
 
-    Assert.assertNull(externalView, "external-view for TestDB0 should be removed, but was: "
-        + externalView);
+    result = TestHelper.verify(() -> {
+      ExternalView externalView = accessor.getProperty(keyBuilder.externalView("TestDB0"));
+      return externalView == null;
+    }, TestHelper.WAIT_DURATION);
+
+    ExternalView ev = accessor.getProperty(keyBuilder.externalView("TestDB0"));
```

Comment:
Why get it once more?

Reply:
If it is just for the assert log, then you can do this,

ExternalView externalView = null;
result = TestHelper.verify(() -> {
  externalView = accessor.getProperty(keyBuilder.externalView("TestDB0"));
  return externalView == null;
}, TestHelper.WAIT_DURATION);
Assert.assertTrue(result, "external-view for TestDB0 should be removed, but was: " + externalView);

Reply:
This way does not work. Lambda varable captured (externalView here) needs to be final. Changed it to use instance variable then.

Reply:
It certainly works. But it would be better to have a wrapper class (AtomicReference for example), to contain the result. The wrapper class can be final. Anyway, the goal is to ensure all test cases are independent of each other as far as we can.
Just a suggestion, the current logic is also fine : )